src/Api/Controllers/UserCreditCardsController.cs

```diff
@@ -0,0 +1,92 @@
+﻿using CloudDrive.Domain.Entities;
+using CloudDrive.Services.CreditCards;
+using Microsoft.AspNetCore.Mvc;
+namespace clouddrive.Controllers
+{
+    [Route("/api/UserCreditCards")]
+    public class UserCreditCardsController : Controller
+    {
+        private readonly ICreditCardsServices _creditCardsService;
+        public UserCreditCardsController(ICreditCardsServices creditCardsService)
+        {
+            _creditCardsService = creditCardsService;
+        }
+      
+        [HttpGet("{id}")]
+        /*
+         * Get CreditCard for the user 
+         */
+        public async Task<IActionResult> GetCreditCard(int id)
+        {
+            try
+            {
+                var userCreditCard = await _creditCardsService.GetUserCreditCardAsync(id);
+                if (userCreditCard == null)
+                {
+                    return NotFound();
+                }
+                else
+                {
+                    return Ok(userCreditCard);
+                }
+            } catch (Exception ex)
+            {
+                return StatusCode(500, ex);
```

Comment:
Please do not return the exception to the user, that is a bad practice, either return a formatted message if the error is known or just return an error code if its not